src/view/dataElements/xdmObject.jsx

```diff
@@ -122,35 +120,27 @@ const XdmExtensionView = () => {
   const [schemasMeta, setSchemasMeta] = useState();
   const [selectedSchemaMeta, setSelectedSchemaMeta] = useState();
 
+  // TODO: break apart this extension view
   return (
     <ExtensionView
       getInitialValues={({ initInfo }) => {
-        // TODO: If we've already loaded the schemas, then don't load the
-        // schemas again. Just call getInitialValues with the newly
-        // selected schema and return the result.
-        return fetchSchemaList({
+        return fetchSchemasMeta({
           orgId: initInfo.company.orgId,
           imsAccess: initInfo.tokens.imsAccess
         })
           .then(_schemasMeta => {
             setSchemasMeta(_schemasMeta);
 
-            if (!_schemasMeta.length) {
+            if (!_schemasMeta || !_schemasMeta.length) {
```

Comment:
Hey @garyr, when do you suspect `_schemasMeta` will be falsey? I'm wondering if this is a case where the promise should be rejected instead of resolved.

Reply:
This might just be an issue better handled elsewhere, but when I have a bad access token, and the fetchSchemasMeta call fails, I believe the variable is null and doesn't have a length property. This causes an unknown length property error, instead of the better error message

Reply:
I updated with better handling of the 401 conditions